plugin/edit.py

```diff
@@ -2,7 +2,20 @@
 import sublime_plugin
 from .core.edit import sort_by_application_order, TextEdit
 from .core.logging import debug
-from .core.typing import List, Dict, Optional, Any
+from .core.typing import List, Dict, Optional, Any, Generator
+from contextlib import contextmanager
+
+
+@contextmanager
+def temporary_view_setting(view: sublime.View, key: str, value: Any) -> Generator[None, None, None]:
+    settings = view.settings()
+    previous_value = settings.get(key)
+    settings.set(key, value)
+    yield
+    if previous_value is None:
+        settings.erase(key)
+    else:
+        settings.set(key, previous_value)
```

Comment:
In case `translate_tabs_to_spaces: true` is set in global preferences, the `settings.get()` will return `True`, not `None`. So then we'll also keep the `True` value set on the view while it wasn't really set initially.

My point: I think it makes sense to always erase. At least we won't leave view-local setting behind on all views that were saved.

Reply:
And I suppose for this example it doesn't matter if `translate_tabs_to_spaces` is set to `True` initially.
It will be always either True or False unless somehow the user is missing default settings.

Reply:
Oh I see what you mean. This might indeed be problematic.

Reply:
I'm fine with this PR if you update this to the version that you've posted on Discord.